plugins/chat/assets/javascripts/discourse/components/chat-channel.js

```diff
@@ -299,7 +299,7 @@ export default class ChatChannel extends Component {
     let foundFirstNew = false;
     const hasNewest = this.messagesManager.messages.some((m) => m.newest);
 
-    result.messages.forEach((messageData, index) => {
+    result?.messages?.forEach((messageData, index) => {
```

Comment:
This is also part of the fix, but was a consequence of the error. If we have an error, result won't have messages, so it was actually failing here instead of showing the real problem. Ultimately this needs to be better in the future.